src/stream-analytics/azext_stream_analytics/_params.py

```diff
@@ -0,0 +1,180 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    resource_group_name_type,
+    get_location_type
+)
+from azure.cli.core.commands.validators import validate_file_or_dict
+from .vendored_sdks.streamanalytics.models import OutputStartMode, EventsOutOfOrderPolicy, OutputErrorPolicy
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('stream-analytics job create') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx))
+        c.argument('tags', tags_type)
+        c.argument('events_out_of_order_policy', arg_type=get_enum_type(EventsOutOfOrderPolicy),
+                   help='Indicates the policy to apply to events that arrive out of order in the input event stream.')
+        c.argument('output_error_policy', arg_type=get_enum_type(OutputErrorPolicy),
+                   help='Indicates the policy to apply to events that arrive at the output.')
+        c.argument('events_out_of_order_max_delay_in_seconds', type=int,
+                   help='The maximum tolerable delay in seconds where out-of-order events can be adjusted to be back in order.')
+        c.argument('events_late_arrival_max_delay_in_seconds', type=int,
+                   help='The maximum tolerable delay in seconds where events arriving late could be included.  Supported range is -1 to 1814399 (20.23:59:59 days) and -1 is used to specify wait indefinitely.')
+        c.argument('data_locale',
+                   help='The data locale of the stream analytics job. Defaults to \'en-US\' if none specified.')
+
+    with self.argument_context('stream-analytics job update') as c:
+        c.argument('resource_group', resource_group_name_type)
+        c.argument('name', options_list=['--name', '-n'], help='The name of the streaming job.')
+        c.argument('tags', tags_type)
+        c.argument('events_out_of_order_policy', arg_type=get_enum_type(EventsOutOfOrderPolicy),
```

Comment:
too long.. event-outoforder-policy?

Reply:
update for both events_outoforder_policy and events_outoforder_max_delay.